x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authc/saml/SamlAuthenticatorTests.java

```diff
@@ -403,7 +403,7 @@ public void testNoAttributesReturnedWhenTheyCannotBeDecrypted() throws Exception
         final String xml = getSimpleResponseAsString(now);
 
         // Encrypting with different cert instead of sp cert will mean that the SP cannot decrypt
-        final Response encrypted = encryptAttributes(xml, readKeyPair("RSA_4096_updated"));
+        final Response encrypted = encryptAttributes(xml, readKeyPair("ENCRYPTION_RSA_4096_updated"));
```

Comment:
I could use some help to understand what is going on here: Is this necessary because `RSA_4096_updated` is used in other tests as a signing key, e.g. in `testSigningKeyIsReloadedForEachRequest`? Since the all tests run in the same JVM, once this key is added as a signing key, it can no longer be used as a encryption key here and vice versa (depends on which test runs first)?

Reply:
Correct!

> which meant that a single key would be
used both for signing and encryption in the same run